v4/export/dump_test.go

```diff
@@ -195,3 +198,36 @@ func (s *testDumpSuite) TestDumpTableWhereClause(c *C) {
 	c.Assert(tbDataRes.Rows().HasNext(), IsFalse)
 	c.Assert(mock.ExpectationsWereMet(), IsNil)
 }
+
+func (s *testDumpSuite) TestDumpDatabaseWithRetry(c *C) {
+	mockConfig := DefaultConfig()
+	mockConfig.SortByPk = false
+	mockConfig.Databases = []string{"test"}
+	mockConfig.Tables = NewDatabaseTables().AppendTables("test", "t")
+	db, mock, err := sqlmock.New()
+	c.Assert(err, IsNil)
+
+	showCreateDatabase := "CREATE DATABASE `test`"
+	rows := mock.NewRows([]string{"Database", "Create Database"}).AddRow("test", showCreateDatabase)
+	mock.ExpectQuery("SHOW CREATE DATABASE `test`").WillReturnRows(rows)
+	showCreateTableResult := "CREATE TABLE t (a INT)"
+	rows = mock.NewRows([]string{"Table", "Create Table"}).AddRow("t", showCreateTableResult)
+	mock.ExpectQuery("SHOW CREATE TABLE `test`.`t`").WillReturnRows(rows)
+	rows = mock.NewRows([]string{"column_name", "extra"}).AddRow("id", "").AddRow("name", "")
+	mock.ExpectQuery("SELECT COLUMN_NAME").WithArgs(sqlmock.AnyArg(), sqlmock.AnyArg()).WillReturnRows(rows)
+	rows = mock.NewRows([]string{"a"}).AddRow(1)
+	mock.ExpectQuery("SELECT (.) FROM `test`.`t` LIMIT 1").WillReturnRows(rows)
+	rows = mock.NewRows([]string{"a"}).AddRow(1).AddRow(2)
+	mock.ExpectQuery("SELECT (.) FROM `test`.`t`").WillReturnError(&mysql.MySQLError{Number: errno.ErrPDServerTimeout, Message: "pd is timeout"})
```

Comment:
can we use `9001` in place of `errno.ErrPDServerTimeout` here? looks like a total waste to add `tidb` to go.mod just to use a single constant in a unit test.